pkg/auth/providers/github/github_client.go

```diff
@@ -76,7 +76,27 @@ func (g *GClient) getUser(githubAccessToken string, config *v3.GithubConfig) (Ac
 		logrus.Errorf("Github getGithubUser: error unmarshalling response, err: %v", err)
 		return Account{}, err
 	}
-
+	// If the UserInfo endpoint does not return the email of a user, we need to query the emails endpoint
+	if githubAcct.Email == "" {
+		emailURL := g.getURL("USER_EMAIL", config)
+		emailResp, _, err := g.getFromGithub(githubAccessToken, emailURL)
+		if err != nil {
+			logrus.Errorf("Github getGithubUser: GET email url %v received error from github, err: %v", emailURL, err)
+			return Account{}, err
+		}
+		var userEmails []EmailResponse
+		if err := json.Unmarshal(emailResp, &userEmails); err != nil {
+			logrus.Errorf("Github getGithubUser: error unmarshalling email response, err: %v", err)
+			return Account{}, err
+		}
+		// This api endpoint doesn't accept query parameter to filter on primary, so we get all emails and loop through it
+		for _, userEmail := range userEmails {
+			if userEmail.Primary {
+				githubAcct.Email = userEmail.Email
+				break
+			}
+		}
```

Comment:
Seems like this might be required by the following as well: https://github.com/rancher/rancher/blob/774dc55f6e1c4dba46906fbbcf42ffaeba336509/pkg/auth/providers/github/github_client.go#L242-L258

Can we create a new function `getUserEmailID` so it can be called by both?

Reply:
I hadn't seen that, and I'm not sure why we call userinfo endpoint for org

Reply:
This is not calling userinfo endpoint for org - It is just a confusing method name. I think is a method to get User Or Org (group) By ID

Reply:
This call should only be returning public info of the user/ since this can be called by user search

Reply:
So if we are going to add in email; it should not be private email. Its probably better to not add email data in here.

Reply:
But I don't think we should get email at all here, since this is not what we store on userattribute. So it's an additional api call that we can avoid

Reply:
Yeah better to not add email data here, since this is not what we store on userattribute. So it's an additional api call that we can avoid